cpp/util/diagonalize.cc

```diff
@@ -3,7 +3,7 @@
 namespace freud { namespace util {
 
 void diagonalize33SymmetricMatrix(const util::ManagedArray<float>& mat, util::ManagedArray<float>& eigen_vals,
-                                  util::ManagedArray<float>& eigen_vecs)
+                                  const util::ManagedArray<float>& eigen_vecs)
```

Comment:
I don't think this is `const`... The eigenvectors are calculated by Eigen and written to this ManagedArray.

Reply:
Lol good catch. I wonder how this got through, somehow it didn't identify that the `get` method is non-const.

Reply:
Yes, so this problem is because we require a non-const pointer in Cython for constructing any views. We also need to use the `get` function internally in a few places. For some reason I guess I decided a good compromise would be to define the method as `const` despite returning a non-const pointer... not sure what I was thinking then. I'm just going to overload the method, and it'll be up to calling code to not break the underlying array with invalid access patterns.

Reply:
Ah now I'm really remembering all the headaches that come up here. The code that Cython generates does not handle the overload gracefully. So we will need to do some slightly more annoying things to make this work.